File: benchmarks/objects_and_arrays/objects_arrays.py
class Person:
    def __init__(self, id, age, weight):
        self.id = id
        self.age = age
        self.weight = weight
    def age_hundred(self):
        while self.age < 100: self.age += 1
        return self
    def inc_weight(self, increment):
        self.weight += increment; return self
    def print(self):
        return f"Person {self.id} - Age {self.age}: Weight {self.weight}"

people = [Person(i, 100 - i, 80000 - i) for i in range(1, 10001)]
for p in people: p.age_hundred().inc_weight(p.id)
for p in people: print(p.print())

File: benchmarks/objects_and_arrays/objects_arrays.cpp
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;

class Person {
public:
    int id, age; float weight;
    Person(int id, int age, float weight) {
        this->id = id; this->age = age; this->weight = weight;
    }
    Person* age_hundred() { while (this->age < 100) this->age += 1; return this; }
    Person* inc_weight(int increment) { this->weight += increment; return this; }
    string print() {
        stringstream result;
        result << "Person " << id << " - Age: " << age << ", Weight: " << weight;
        return result.str();
    }
};
int main() {
    vector<Person> people;
    for (int i = 1; i <= 10000; i++) people.push_back(Person(i, 100 - i, 80000 - i));
    for (Person p: people) p.age_hundred()->inc_weight(p.id);
    for (Person p: people) cout << p.print() << endl;
}
